lib/fluent/plugin/in_tail.rb

```diff
@@ -248,6 +248,10 @@ def stop_watchers(paths, immediate = false, unwatched = false)
 
     # refresh_watchers calls @tails.keys so we don't use stop_watcher -> start_watcher sequence for safety.
     def update_watcher(path, pe)
+      unless pe.read_inode == @pf[path].read_inode
```

Comment:
Maybe, `@pf` check is needed?
@pf is `nil` when `pos_file` option is not specified.